coffee_core/src/coffee.rs

```diff
@@ -330,6 +330,15 @@ impl PluginManager for CoffeeManager {
         );
         Err(err)
     }
+
+    async fn search(&mut self, plugin: &str) -> Result<Value, CoffeeError> {
+        for repo in &self.repos {
+            if let Some(_plugin) = repo.get_plugin_by_name(plugin) {
+                return Ok(json!({ "search": repo.name() }));
+            }
+        }
+        return Ok(json!({ "search": "not found" }));
```

Comment:
```suggestion
        return Err(CoffeErrro::new(404, format!("plugin `{plugin}` not found")))
```

Still wrong, this is the correct way to do it